moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -296,129 +285,58 @@ moveit::core::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const
-            }
-          }
-
-          // make sure there is a default resolution at least specified for every solver (in case it was not specified
-          // on the param server)
-          while (search_res[known_group.name_].size() < possible_kinematics_solvers[known_group.name_].size())
-            search_res[known_group.name_].push_back(default_search_resolution_);
-        }
-      }
-      else
-      {
-        RCLCPP_DEBUG(LOGGER, "Using specified default settings for kinematics solvers ...");
-        for (const srdf::Model::Group& known_group : known_groups)
-        {
-          possible_kinematics_solvers[known_group.name_].resize(1, default_solver_plugin_);
-          search_res[known_group.name_].resize(1, default_search_resolution_);
-          ik_timeout_[known_group.name_] = default_solver_timeout_;
-          groups_.push_back(known_group.name_);
-        }
+        groups_.push_back(known_group.name_);
+
+        std::string kinematics_param_prefix = robot_description_ + "_kinematics." + known_group.name_;
+        kinematics_param_listener_.try_emplace(
+            known_group.name_, std::make_shared<kinematics::ParamListener>(node_, kinematics_param_prefix));
+        kinematics_params_.try_emplace(known_group.name_,
+                                       kinematics_param_listener_.at(known_group.name_)->get_params());
+
+        std::string kinematics_solver_param_name = kinematics_param_prefix + ".kinematics_solver";
+        const auto kinematics_solver = kinematics_params_.at(known_group.name_).kinematics_solver;
+        possible_kinematics_solvers[known_group.name_].push_back(kinematics_solver);
+        RCLCPP_DEBUG(LOGGER, "Using kinematics solver '%s' for group '%s'.", kinematics_solver.c_str(),
+                     known_group.name_.c_str());
+
+        std::string kinematics_solver_timeout_param_name = kinematics_param_prefix + ".kinematics_solver_timeout";
+        const auto kinematics_solver_timeout = kinematics_params_.at(known_group.name_).kinematics_solver_timeout;
+        RCLCPP_DEBUG(LOGGER, "Param %s : %f", kinematics_solver_timeout_param_name.c_str(), kinematics_solver_timeout);
+        ik_timeout_[known_group.name_] = kinematics_solver_timeout;
+
+        std::string kinematics_solver_res_param_name = kinematics_param_prefix + ".kinematics_solver_search_resolution";
+        const auto kinematics_solver_search_resolution =
+            kinematics_params_.at(known_group.name_).kinematics_solver_search_resolution;
+        RCLCPP_DEBUG(LOGGER, "Param %s : %f", kinematics_solver_res_param_name.c_str(),
+                     kinematics_solver_search_resolution);
+        search_res[known_group.name_].push_back(kinematics_solver_search_resolution);
+
+        // Allow a kinematic solver's tip links to be specified on the rosparam server as an array
+        std::string ksolver_ik_links_param_name = kinematics_param_prefix + ".kinematics_solver_ik_links";
```

Comment:
This parameter is not set anywhere and I'm not sure if this is being used. Should I keep this remove this parameter check?